Added Minesweeper generator

felix91gr opened the pull request:
> - Added generator itself
> - Updated generated Test file
> - Fixed Example and Minesweeper classes to recieve and return the right type on method Annotate.
  Was: string -> string. Now it is: string[] -> string[]

@ErikSchierboom please tell me I didn't screw up by changing `Annotate` from a `string -> string` function to a `string[] -> string[]` one? I don't know how else to interpret the Canonical Data [files](https://github.com/exercism/problem-specifications/blob/master/exercises/minesweeper/canonical-data.json).

If that's allright, then this commit is pretty much cooked afaik.

felix91gr commented:
I branched this one off of `master`, so it's clean to be merged before #471 and #412

felix91gr commented:
I'm done with number 1.

> Format the arrays such that they are neatly aligned on different lines.

Now I'm on to this.

felix91gr commented:
@ErikSchierboom this one's ready as well ;)

ErikSchierboom commented on exercises/minesweeper/MinesweeperTest.cs:
```diff
@@ -1,142 +1,240 @@
     {
-        var input = "";
-        var expected = "";
+        var input = new [] 
```
I think no rows should mean an empty array, not an array with an empty string. Could you replace that (also for the expected value)?

felix91gr replied:
Sure!

felix91gr commented:
Okay, now I've fixed and tested it. The arrays it creates now are actually better, because they are strictly `new string[]`. 

Creating just `new []` is not a good idea imo, if we can avoid it. In this case, we _know_ that the input is exactly a `string` array, so I changed that as well.

Thanks for the feedback.
Tell me what you think of this one :)

ErikSchierboom commented:
Merged. Thanks! :tada:
